SimTracker/TrackTriggerAssociation/interface/TTTrackAssociationMap.h

```diff
@@ -0,0 +1,121 @@
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/Common/interface/DetSet.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2TrackerDigi.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/MeasurementPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h" /// NOTE: this is needed even if it seems not
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "SimDataFormats/TrackerDigiSimLink/interface/PixelDigiSimLink.h"
+#include "SimDataFormats/Track/interface/SimTrack.h"
+#include "SimDataFormats/Track/interface/SimTrackContainer.h"
+#include "SimDataFormats/EncodedEventId/interface/EncodedEventId.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
+#include "SimTracker/TrackTriggerAssociation/interface/TTStubAssociationMap.h"
+
+template< typename T >
+class TTTrackAssociationMap
+{
+  public:
+    /// Constructors
+    TTTrackAssociationMap();
+
+    /// Destructor
+    ~TTTrackAssociationMap();
+
+    /// Data members:   getABC( ... )
+    /// Helper methods: findABC( ... )
+
+    /// Maps
+    std::map< edm::Ptr< TTTrack< T > >, edm::Ptr< TrackingParticle > >                getTTTrackToTrackingParticleMap() const
+      { return trackToTrackingParticleMap; }
+    std::map< edm::Ptr< TrackingParticle >, std::vector< edm::Ptr< TTTrack< T > > > > getTrackingParticleToTTTracksMap() const
+      { return trackingParticleToTrackVectorMap; }
+
+    void setTTTrackToTrackingParticleMap( std::map< edm::Ptr< TTTrack< T > >, edm::Ptr< TrackingParticle > > aMap )
+      { trackToTrackingParticleMap = aMap; }
+    void setTrackingParticleToTTTracksMap( std::map< edm::Ptr< TrackingParticle >, std::vector< edm::Ptr< TTTrack< T > > > > aMap )
+      { trackingParticleToTrackVectorMap = aMap; }
+    void setTTStubAssociationMap( edm::RefProd< TTStubAssociationMap< T > > aStubAssoMap )
+      { theStubAssociationMap = aStubAssoMap; }
+
+    /// Operations
+    edm::Ptr< TrackingParticle >            findTrackingParticlePtr( edm::Ptr< TTTrack< T > > aTrack ) const;
+    std::vector< edm::Ptr< TTTrack< T > > > findTTTrackPtrs( edm::Ptr< TrackingParticle > aTrackingParticle ) const; 
+
+    /// MC Truth methods
+    bool isGenuine( edm::Ptr< TTTrack< T > > aTrack ) const;
+    bool isLooselyGenuine( edm::Ptr< TTTrack< T > > aTrack ) const;
+    bool isCombinatoric( edm::Ptr< TTTrack< T > > aTrack ) const;
+    bool isUnknown( edm::Ptr< TTTrack< T > > aTrack ) const;
+
+  private:
+    /// Data members
+    std::map< edm::Ptr< TTTrack< T > >, edm::Ptr< TrackingParticle > >                trackToTrackingParticleMap;
+    std::map< edm::Ptr< TrackingParticle >, std::vector< edm::Ptr< TTTrack< T > > > > trackingParticleToTrackVectorMap;
```

Comment:
Depending on the expected number of entries in these maps, it might be preferable to use unordered_maps. (Doesn't need to be done in this PR, just a suggestion for the future.)